go.mod

```diff
@@ -72,13 +80,16 @@ require (
 	gomodules.xyz/jsonpatch/v2 v2.5.0 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20250218202821-56aae31c358a // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20250528174236-200df99c418a // indirect
+	google.golang.org/grpc v1.72.0 // indirect
 	google.golang.org/protobuf v1.36.6 // indirect
 	gopkg.in/evanphx/json-patch.v4 v4.12.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	k8s.io/apiextensions-apiserver v0.33.1 // indirect
 	k8s.io/klog/v2 v2.130.1 // indirect
 	k8s.io/kube-openapi v0.0.0-20250318190949-c8a335a9a2ff // indirect
+	kcl-lang.io/kcl-go v0.11.2 // indirect
```

Comment:
I'd prefer to keep these dependencies out of the main module since they aren't used by Eno itself and might cause conflicts or just dependency bloat when folks import the synthesizer SDK (pkg/function).

The project uses go workspaces so you should be able to initialize a new go module in pkg/kclshim and add it to the workspace. Helmshim already does this today